### DISCUSSION TITLE
Build OTBR firmwre with TREL enabled

### DISCUSSION THREAD

**Discussion by GhitaRk** [openthread/openthread#8884] [Q&A] [answered]
Hello,

I am using Silicon Labs EFR32MG12 boards and I have two thread networks, with two border routers connected via ethernet.
Now I want to connect them using TREL. The problem is that I am not able to build firmwares with TREL. Theses two commandes failed:
./script/build "brd4161a" -DOT_TREL=ON
./script/build "brd4161a" -DOT_APP_NCP=ON -DOT_TREL=ON

But I was able to build and flash RCP binary with TREL enabled using Simplicity studio. However I still have this error:
pi@raspberrypi:~ $ sudo ot-ctl trel
Error 35: InvalidCommand

I also tried  ./script/cmake-build -DOTBR_TREL=ON in RaspberryPi ot-br-posix project, it was build without any problem but I don't know if I should flash any binary after building.

What should I do more to enable TREL. Does Silabs boards support TREL ?

**Comment by silabs-grace** [comment, marked as answer]
TREL is supported by Silicon Labs boards. I would like to point out that for the Silicon Labs Solution it is important to make sure that the border router and the RCP are on matching commit versions to ensure proper interoperability. Furthermore the OTBR and RCP need to both be built from the same sources & matching versions; either both from the GSDK or the Github repos. This will help minimize the likelihood of incompatibilities that may result from combinations that are not supported. Information on the compatible commit versions can be found in the Open Thread Release Notes section 8.2 [here](https://www.silabs.com/documents/public/release-notes/open-thread-release-notes-2.2.2.0.pdf#8.2%20OpenThread%20GitHub%20Repository). All of our release notes can be found here:  https://www.silabs.com/developers/gecko-software-development-kit.

Information on how to build the OTBR with TREL enabled is explained in [AN1372 Section 2 Table 2-1](https://www.silabs.com/documents/public/application-notes/an1372-configuring-for-thread-1-3.pdf#page=4). Essentially you will need to set the OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE build option as well as configure the TREL infrastructure link in the radio URL on the OTBR.
